packages/dx-grid-core/src/plugins/table-band-header/helpers.js

```diff
@@ -31,87 +32,158 @@ export const getColumnMeta = (
   return acc;
 }, result || { level, title });
 
+export const getCurrentColumnBandInfo = (
+  { currentColumnIndex, currentTableColumn, currentRowLevel },
+  { tableColumns, columnBands },
+) => {
+  const currentRealTableColumn = tableColumns[currentColumnIndex];
+  const columnMayBeInBand = currentTableColumn.type === TABLE_DATA_TYPE
+    || (currentTableColumn.type === TABLE_STUB_TYPE
+      && currentRealTableColumn.type === TABLE_DATA_TYPE);
+
+  return columnMayBeInBand
+    ? getColumnBandInfo(currentRealTableColumn.column.name, columnBands, currentRowLevel)
+    : { level: 0, title: '' };
+};
+
+const getColumnIndex = (currentTableColumn, tableColumns) => (
+  currentTableColumn.type === TABLE_STUB_TYPE
+    ? currentTableColumn.rightColumnIndex
+    : tableColumns.findIndex(column => column.key === currentTableColumn.key)
+);
+
+const bandCellWillBeMerged = (
+  {
+    currentTableColumn, currentColumnIndex,
+    currentColumnBand, currentRowLevel,
+    previousTableColumn, isCurrentColumnFixed,
+  },
+  columnBands,
+) => {
+  if (currentColumnIndex <= 0 || previousTableColumn.type !== TABLE_DATA_TYPE) return false;
+
+  const isPrevColumnFixed = !!previousTableColumn.fixed;
+  const prevColumnBand = getColumnBandInfo(
+    previousTableColumn.column.name,
+    columnBands,
+    currentRowLevel,
+  );
+
+  return (prevColumnBand.title === currentColumnBand.title
+    && !currentTableColumn.rightColumnIndex
+    && (!isPrevColumnFixed || (isPrevColumnFixed && isCurrentColumnFixed)));
+};
+
+const columnRenderedInOtherRow = ({ currentColumnBand, currentRowLevel }) => (
+  currentColumnBand.level < currentRowLevel
+);
+
+const columnBelongsToCurrentRow = ({ currentColumnBand, currentRowLevel }) => (
```

Comment:
Can we rename this functions similarly? I mean `columnRenderedInOtherRow` and (`columnBelongsToCurrentRow` => `columnRenderedInCurrentRow`)

Reply:
these functions contain slightly different logic: 
`columnRenderedInOtherRow` means that a current header cell has _already_ been rendered in some upper row (and we replace it with empty cell), 
`columnBelongsToCurrentRow` returns true for a cell that has not been yet rendered.
We can consider names like `shouldRenderColumnInCurrentRow`, `isColumnInCurrentRow` or something similar

Reply:
I prefer `isColumnInCurrentRow` or `isCurrentRowColumn`

Reply:
`shouldRenderColumnInCurrentRow` and `isColumnInCurrentRow`

+1